puppetdb.go

```diff
@@ -3,38 +3,42 @@ package autosignr
 import (
 	"crypto/tls"
 	"encoding/json"
-	"errors"
 	"fmt"
 	"io/ioutil"
 	"net/http"
+	"strings"
 	"time"
+
+	"github.com/pkg/errors"
 )
 
+// PuppetDBNode - Returning from the API call to puppetDB
 type PuppetDBNode struct {
-	Certname string
+	Certname string `json:"certname"`
 }
 
-// Queries PuppetDB and returns the list of nodes found to be inactive, where
+// FindInactiveNodes Queries PuppetDB and returns the list of nodes found to be inactive, where
 // inactive is the last report timestamp is greater than $hours old
-func FindInactiveNodes(hours int, host string, protocol string, uri string, ignore_cert_errors bool) ([]string, error) {
-	var list []string
-
+func FindInactiveNodes(hours int, host string, protocol string, uri string, ignoreCertErrors bool, includeFilters []string) ([]string, error) {
 	t := time.Now().Add(time.Hour * time.Duration(hours*-1)).Format(time.RFC3339)
 
-	url := fmt.Sprintf("%s://%s%s?query=[\"<\",\"report_timestamp\",\"%s\"]",
-		protocol,
-		host,
-		uri,
-		t)
+	url := fmt.Sprintf("%s://%s%s", protocol, host, uri)
+
+	data := fmt.Sprintf("{ \"query\": \"nodes[certname]{ report_timestamp < \\\"%s\\\"", t)
+	for _, val := range includeFilters {
```

Comment:
perhaps swap this loop for a `strings.Join(includeFilters, " ")` instead, then inline into the sprintf?

```
data := fmt.Sprintf(
    "{ \"query\": \"nodes[certname]{ report_timestamp < \\\"%s\\\" %s }\"}",
    t,
    strings.Join(includeFilters, " "),
)
```